src/ccalls.jl

```diff
@@ -130,9 +130,7 @@ end
 > There is no argument number checking here, all checks on the arguments
 > are done by GAP itself.
 """
-# # The (func::GapObj) function (commented out below) needs to be instantiated for `MPtr` 
-# # and is therefore moved to the init function
-# (func::GapObj)(args...; kwargs...) where T <: GapObj = call_gap_func(func, args...; kwargs...)
+(func::GapObj)(args...; kwargs...) where T <: GapObj = call_gap_func(func, args...; kwargs...)
```

Comment:
@kalmarek in our Skype just now, I think you were trying to point out some issue with the above code to me (in terms of efficiency perhaps?), but I didn't quite follow; if you could explain here, that'd be much appreciated :-)

Reply:
I actually thought of
```julia
(func::T)(args...; kwargs...) where T <: GAP.GapObj = call_gap_func(func, args...; kwargs...)
```
but it just doesn't work :P with `ERROR: function type in method definition is not a type`.
Sorry for bothering you ;)

Reply:
if we stick with `GAPTypes` `GapObj` we need to wait for 1.3 to define
```julia
(func::GapObj)(args...; kwargs...) = call_gap_func(func, args...; kwargs...)
```

We could also try 
```julia
struct GapObj{T}
   mptr{T}
end
```
so that `MPtr` doesn't need a supertype and we can write
```julia
(func::GapObj)(args...;kwargs...) = call_gap_func(func.mptr, args..., kwargs...)
```